File: kt/src/main/kotlin/io/github/ephemient/aoc2020/Day25.kt
package io.github.ephemient.aoc2020

class Day25(lines: List<String>) {
    private val pub1 = lines[0].toInt()
    private val pub2 = lines[1].toInt()

    fun part1(): Int {
        var e = 0
        var n = 1
        while (n != pub2) {
            e++
            n = 7 * n % 20201227
        }
        return pub1.toBigInteger().modPow(e.toBigInteger(), 20201227.toBigInteger()).intValueExact()
    }
}

File: py/src/aoc2020/day25.py
import fileinput


def part1(lines):
    '''
    >>> part1(('5764801', '17807724'))
    14897079
    '''
    pub1, pub2 = [*map(int, lines)]
    e, n = 0, 1
    while n != pub2:
        e += 1
        n = 7 * n % 20201227
    return pow(pub1, e, 20201227)


parts = (part1, )

if __name__ == '__main__':
    lines = list(fileinput.input())
    print(part1(lines))
